File: library/src/main/java/br/com/mauker/materialsearchview/lib/db/HistoryContract.java
package br.com.mauker.materialsearchview.lib.db;

import android.content.ContentUris;
import android.net.Uri;
import android.provider.BaseColumns;

/**
 * Created by mauker on 15/04/16.
 * Database contract. Contains the definition of my tables.
 */
public class HistoryContract {

    public static final String CONTENT_AUTHORITY = "br.com.mauker.materialsearchview.searchhistorydatabase";

    private static final Uri BASE_CONTENT_URI = Uri.parse("content://" + CONTENT_AUTHORITY);

    public static final String PATH_HISTORY = "history";

    // ----- Table definitions ----- //

    public static final class HistoryEntry implements BaseColumns {
        // Content provider stuff.
        public static final Uri CONTENT_URI =
                BASE_CONTENT_URI.buildUpon().appendPath(PATH_HISTORY).build();

        public static final String CONTENT_TYPE =
                "vnd.android.cursor.dir/" + CONTENT_URI + "/" + PATH_HISTORY;

        public static final String CONTENT_ITEM =
                "vnd.android.cursor.item/" + CONTENT_URI + "/" + PATH_HISTORY;

        // Table definition stuff.
        public static final String TABLE_NAME = "SEARCH_HISTORY";

        public static final String COLUMN_QUERY = "query";
        public static final String COLUMN_INSERT_DATE = "insert_date";
        public static final String COLUMN_IS_HISTORY = "is_history";

        public static Uri buildHistoryUri(long id){
            return ContentUris.withAppendedId(CONTENT_URI, id);
        }
    }

    // ----- !Table definitions ----- //

}

File: library/src/main/java/br/com/mauker/materialsearchview/db/HistoryContract.kt
package br.com.mauker.materialsearchview.db

import android.content.ContentUris
import android.net.Uri
import android.provider.BaseColumns

/**
 * Created by mauker on 15/04/16.
 * Database contract. Contains the definition of my tables.
 */
object HistoryContract {
    @JvmField
    val CONTENT_AUTHORITY = initAuthority()
    private val BASE_CONTENT_URI = Uri.parse("content://$CONTENT_AUTHORITY")
    const val PATH_HISTORY = "history"

    // ----- Authority setup ----- //
    private fun initAuthority(): String {
        var authority = "br.com.mauker.materialsearchview.defaultsearchhistorydatabase"
        try {
            val clazzLoader = HistoryContract::class.java.classLoader
            val clazz = clazzLoader.loadClass("br.com.mauker.MsvAuthority")
            val field = clazz.getDeclaredField("CONTENT_AUTHORITY")
            authority = field[null].toString()
        } catch (e: ClassNotFoundException) {
            e.printStackTrace()
        } catch (e: NoSuchFieldException) {
            e.printStackTrace()
        } catch (e: IllegalAccessException) {
            e.printStackTrace()
        }
        return authority
    }

    // ----- Table definitions ----- //
    object HistoryEntry {
        // Content provider stuff.
        @JvmField
        val _COUNT: String? = "_count"
        @JvmField
        val _ID = "_id"
        @JvmField
        val CONTENT_URI = BASE_CONTENT_URI.buildUpon().appendPath(PATH_HISTORY).build()
        @JvmField
        val CONTENT_TYPE = "vnd.android.cursor.dir/$CONTENT_URI/$PATH_HISTORY"
        @JvmField
        val CONTENT_ITEM = "vnd.android.cursor.item/$CONTENT_URI/$PATH_HISTORY"

        // Table definition stuff.
        const val TABLE_NAME = "SEARCH_HISTORY"
        const val COLUMN_QUERY = "query"
        const val COLUMN_INSERT_DATE = "insert_date"
        const val COLUMN_IS_HISTORY = "is_history"
        @JvmStatic
        fun buildHistoryUri(id: Long): Uri {
            return ContentUris.withAppendedId(CONTENT_URI, id)
        }
    } // ----- !Table definitions ----- //
}